platform/jewel/ui/src/main/kotlin/org/jetbrains/jewel/ui/component/Menu.kt

```diff
@@ -239,7 +256,10 @@ private fun MenuItem(item: MenuItem, showIcons: Boolean, showKeybindings: Boolea
                     canShowIcon = showIcons,
                     canShowKeybinding = showKeybindings,
                     enabled = item.isEnabled,
-                    content = item.content,
+                    content = {
+                        LaunchedEffect(it.isHovered) { if (it.isHovered) currentSetSelectedSubMenu(null) }
```

Comment:
I don't love (ab)using state this way. Maybe a local function on L247 could also work instead?

Reply:
Added `deselectSubmenu` and changed to run inside a remember